modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/visualization_modes/components/AddFieldModalContent.tsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayModal from '@clayui/modal';
+import classNames from 'classnames';
+import React, {useState} from 'react';
+
+import RequiredMark from '../../../components/RequiredMark';
+import ValidationFeedback from '../../../components/ValidationFeedback';
+import {IField} from '../../../utils/types';
+
+const AddFieldModalContent = ({
+	closeModal,
+	namespace,
+	onSaveButtonClick,
+}: {
+	closeModal: Function;
+	namespace: string;
+	onSaveButtonClick: (selectedField: IField) => void;
+}) => {
+	const [fieldName, setFieldName] = useState<string>();
+	const [
+		requiredFieldNameValidationError,
+		setRequiredFieldNameValidationError,
+	] = useState<boolean>();
```

Comment:
We typically set default on validations to `false`, so that modal "resets" if it is reopened after close on [x]. Any reason that we are not doing it here?

Reply:
Nothing, only using the default value on declaration